src/atn-editor/store/types.ts

```diff
@@ -0,0 +1,98 @@
+import type { ATNValueType } from '../types';
+
+// Element Types
+type ElementId = string;
+
+type BaseElementType = {
+  id: ElementId;
+  name: string;
+  color: string;
+  parentId?: string;
+  defaultAtn: ATNValueType;
+  staticElement: boolean;
+};
+
+type ElementType = BaseElementType & {
+  atn: ATNValueType;
+};
+
+type DisplayIndexType = {
+  [ElementId: string]: number;
+};
+
+type RootOrderType = ElementId[];
+
+type ElementsType = {
+  [ElementId: string]: ElementType;
+};
```

Comment:
This is an [index signature](https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures) right?
Does it have any benefit here?
Wouldn't using `Record<>` make it way easier to read?
```suggestion
type DisplayIndexType = Record<string, number>
type RootOrderType = ElementId[];
type ElementsType = Record<string, ElementType>
```

Reply:
I think it's better to use Record instead. Thanks for the suggestion.